model/LtiListenerService.php

```diff
@@ -125,6 +126,27 @@ public function executionStateChanged(DeliveryExecutionState $event)
     }
 
     /**
+     * @param DeliveryRunEvent $event
+     */
+    public function deliveryRun(DeliveryRunEvent $event)
+    {
+        $delivery = $event->getDelivery();
+        $user = $event->getUser();
+        $remoteLink = \taoLti_models_classes_LtiService::singleton()->getLtiSession()->getLtiLinkResource();
+        $executions = LTIDeliveryTool::singleton()->getLinkedDeliveryExecutions($delivery, $remoteLink, $user->getIdentifier());
+        if (!empty($executions)) {
+            $deliveryExecutionService = $this->getServiceManager()->get(LtiDeliveryExecutionService::SERVICE_ID);
+            foreach ($executions as $deliveryExecution) {
+                if (!$deliveryExecutionService->isFinished($deliveryExecution) && $deliveryExecution->getState()->getUri() != DeliveryExecution::STATE_PAUSED) {
+                    /** @var DeliveryExecutionStateService $deliveryExecutionStateService */
+                    $deliveryExecutionStateService = $this->getServiceManager()->get(DeliveryExecutionStateService::SERVICE_ID);
+                    $deliveryExecutionStateService->pauseExecution($deliveryExecution);
```

Comment:
I guess this solution is not optimal.
All the active executions will be paused. This may be not safe.
Also i don't like that this is event based functionality. This may be not reliable.

Please look at this draft:
https://github.com/oat-sa/extension-tao-ltideliveryprovider/compare/develop...fix/pause_authorized_execution_during_lti_launch?expand=1

You may add checking of state before pausing there. Also this approach requires changes in only one extension.